server/pulp/server/db/querysets.py

```diff
@@ -84,3 +90,26 @@ def update_one(self, *args, **kwargs):
             raise NotImplementedError("The 'multi' parameter cannot be set on this method.")
         kwargs['multi'] = False
         self.update(*args, **kwargs)
+
+
+class RepoQuerySet(CriteriaQuerySet):
+    """
+    Custom queryset for repositories.
+    """
+
+    def get_repo_or_missing_resource(self, repo_id):
```

Comment:
Seems odd to add a whole method like this  as opposed to teaching the pulp middleware about the DoesNotExist exception.